### PR TITLE
Add elasticache_group module

### PR DESCRIPTION
**Pull request by Anislav**
elasticache_group module maintain ElastiCache cache security groups and enables authorization of network ingress for EC2 security groups.

### CODE REVIEW

**Comment by mpdehaan** [comment]
@willthames @pas256 (or others) - care to help review this one?

Thanks!

File: library/cloud/elasticache_group [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+DOCUMENTATION = '''
+---
+module: elasticache_group
+short_description: Maintain an ElastiCache cache security group
+description:
+    - Create or delete an ElastiCache cache security group.
+    - Authorize network ingress for EC2 security groups.
+    - Returns information about the specified ElastiCache cache security group.
+    - This module has a dependency on python-boto >= 2.5.
+version_added: "1.5"
```
Comment (willthames) [inline library/cloud/elasticache_group:12]: Should be 1.6

File: library/cloud/elasticache_group [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+            name=dict(required=True),
+            description=dict(required=True),
+            ec2_security_groups=dict(required=False, type='list'),
+            state=dict(default='present', choices=['present', 'absent']),
+        )
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
```
Comment (willthames) [inline library/cloud/elasticache_group:240]: What does this module do when check mode is switched on?

File: library/cloud/elasticache_group [hunk 3]
```diff
@@ -0,0 +1,270 @@
...
+            info['data'] = self.data
+
+        return info
+
+def main():
+
+    argument_spec = ec2_argument_spec()
+    argument_spec.update(dict(
+            name=dict(required=True),
+            description=dict(required=True),
```
Comment (willthames) [inline library/cloud/elasticache_group:232]: Probably not required when state is absent

File: library/cloud/elasticache_group [hunk 4]
```diff
@@ -0,0 +1,270 @@
...
+        self.changed = False
+        self.conn = self._get_connection()
+
+        self.data = self._describe()
+        self.exists = self.data is not None
+
+    def _get_connection(self):
+        """Get an ElastiCache connection"""
+        try:
+            conn = boto.elasticache.connect_to_region(self.region,
```
Comment (willthames) [inline library/cloud/elasticache_group:110]: Recommend using module_utils/ec2.py `connect_to_aws` as that gives additional support for security_token, profile, validate_certs etc.

File: library/cloud/elasticache_group [hunk 5]
```diff
@@ -0,0 +1,270 @@
...
+
+    if not region:
+        module.fail_json(msg=str("Either region or EC2_REGION environment variable must be set."))
+
+    name = module.params['name']
+    description = module.params['description']
+    state = module.params.get('state')
+    ec2_security_groups = module.params.get('ec2_security_groups')
+
+    group_manager = ElastiCacheGroupManager(module, name, description, ec2_security_groups,
```
Comment (willthames) [inline library/cloud/elasticache_group:253]: Given most of these parameters can be derived from module, I wouldn't have a seven parameter variable list! I'd handle the module.params unpacking in the receiving method or do more in main.

**Comment by willthames** [comment]
In general I'm not a big fan of class based Ansible modules - modules like ec2 and ec2_vol etc seem to be much more readable and writable than e.g. ec2_elb. 

Other than my inline notes, the module itself does seem well written - I haven't tested it though.

**Comment by jimsmith** [comment]
Hi Guys - I need to be able to provision a both memcache and redis within AWS ElastiCache in a VPC - I stumbled across this pull request.

Is there going to be a cloud library for the actual Cache Parameter Groups as this needs to be provisioned prior to creating an ElastiCache resource within AWS VPC ?

**Comment by mpdehaan** [comment]
Hi!

Thanks very much for your interest in Ansible.  It sincerely means a lot to us. 

On September 26, 2014, due to enormous levels of contribution to the project Ansible decided to reorganize module repos, making it easier
for developers to work on the project and for us to more easily manage new contributions and tickets.

We split modules from the main project off into two repos, http://github.com/ansible/ansible-modules-core and http://github.com/ansible/ansible-modules-extras

If you still would like this pull request merged, we will need your help making this target the new repo.  If you do not take any action, this
pull request unfortunately cannot be applied.

We apologize that we are not able to make this transition happen seamlessly, though this is a one-time change and your help is greatly appreciated -- 
this will greatly improve velocity going forward.

Both sets of modules will ship with Ansible, though they'll receive slightly different ticket handling.

To locate where a module lives between 'core' and 'extras'
- Find the module at http://docs.ansible.com/list_of_all_modules.html
- Open the documentation page for that module
- If the bottom of the docs say "This is an extras module", submit your ticket to https://github.com/ansible/ansible-modules-extras
- Otherwise, submit your pull request to update the existing module to https://github.com/ansible/ansible-modules-core
- Note that python modules in ansible now also end in ".py" and this extension is required for new contributions.
- action_plugins (modules with server side components) still live in the main repo.  If your pull request touches both, which should be
   exceedingly rare, submit two new pull requests and make sure to mention the links to each other in the comments.

Otherwise, if this is a new module:
- Submit your pull request to add a module to https://github.com/ansible/ansible-modules-extras

It may be possible to re-patriate your pull requests automatically, one user-submitted approach for advanced git users
has been suggested at https://gist.github.com/willthames/afbaaab0c9681ed45619

Additionally, should you need more help with this, you can ask questions on:
- the development mailing list: https://groups.google.com/forum/#!forum/ansible-devel

Thank you very much!

**Pull request closed without merging**